nighthawk/test/statistic_test.cc

```diff
@@ -0,0 +1,85 @@
+#include <typeinfo> // std::bad_cast
+
+#include "gtest/gtest.h"
+
+#include "nighthawk/common/statistic.h"
+#include "nighthawk/source/common/statistic_impl.h"
+
+namespace Nighthawk {
+
+using MyTypes = ::testing::Types<InMemoryStatistic, HdrStatistic, StreamingStatistic>;
+
+template <typename T> class TypedStatisticTest : public testing::Test {};
+
+class Helper {
+public:
+  /**
+   * With 0 significant digits passed, this uses EXPECT_DOUBLE_EQ. Otherwise EXPECT_NEAR
+   * will be called with a computed acceptable range based on the number of significant
+   * digits and tested_value.
+   * @param expected_value the expected value
+   * @param tested_value the tested_value
+   * @param significant the number of significant digits that should be used to compare values.
+   */
+  static void expect_near(double expected_value, double tested_value, uint64_t significant) {
+    if (significant > 0) {
+      EXPECT_NEAR(expected_value, tested_value,
+                  std::pow(10, std::ceil(std::log10(tested_value)) - 1 - significant));
+    } else {
+      EXPECT_DOUBLE_EQ(expected_value, tested_value);
+    }
+  }
+
+private:
+  Helper() = default;
+};
+
+TYPED_TEST_SUITE(TypedStatisticTest, MyTypes);
```

Comment:
What might be cool is a test that runs all three types with the same data and compares their outputs! :)

Reply:
Also, maybe throwing in some some different pseudo-random examples would be nice bonus. This seems the sort of thing we can mechanically verify across implementations without a lot of manual test creation.